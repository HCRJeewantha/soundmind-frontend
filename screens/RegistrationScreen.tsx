import React, { useState } from "react";
import { Pressable, StyleSheet, TextInput, Text } from "react-native";
import { View } from "../components/Themed";
import { LinearGradient } from 'expo-linear-gradient';
import axios from 'axios'
import Toast from 'react-native-toast-message';

const baseUrl = 'http://127.0.0.1:8000';

export function RegistrationScreen({ navigation }: any) {

    const [userName, setUserName] = useState('');
    const [mobile, setMobile] = useState('');
    const [password, setPassword] = useState('');
    const [confirmPassword, setConfirmPassword] = useState('');

    const showToast = (title: any, message: any, type: any) => {
        Toast.show({
            type: type,
            text1: title,
            text2: message
        });
    }

    const signUp = async () => {
        const payload = {
            "name": userName,
            "mobile": mobile,
            "password": password,
            "confirm_password": confirmPassword
        }

        axios.post(
            baseUrl + '/sign-up',
            payload
        ).then((response) => {
            localStorage.setItem("id", response.data.id)
            localStorage.setItem("name", response.data.name)
            showToast("Success", "Login Successful", "success")
            navigation.navigate('Welcome')
        }).catch((error) => {
            if(error.response){
                showToast("Error", error.response.data.detail, "error")
            }
        });
    };

    const ButtonSignUp = (props: any) => {
        return (
            <Pressable
                style={{
                    alignItems: 'center',
                    justifyContent: 'center',
                    width: '100%',
                }}
                onPress={() => signUp()}>
                <LinearGradient
                    colors={['#50aeffeb', '#26ced7f2']}
                    style={{
                        marginTop: 30,
                        borderRadius: 10,
                        borderWidth: 1,
                        borderColor: '#2196f3',
                        alignItems: 'center',
                        justifyContent: 'center',
                        width: '80%',
                        height: '50px',
                        backgroundColor: '#2196f3',
                        shadowOpacity: 0.25,
                        shadowRadius: 30,
                        shadowOffset: {
                            height: 0,
                            width: 0,
                        },
                    }}>

                    <Text style={{ color: '#fff', fontSize: 20, fontWeight: '600' }}>{props.buttonText}</Text>
                </LinearGradient>
            </Pressable>
        )
    }

    return (
        <View style={styles.container}>
            <LinearGradient
                colors={['#262626', '#75dadf']}
                style={styles.background}
            />
            <View style={{ width: '100%', height: 'auto', bottom: 0, position: 'absolute', backgroundColor: '#ffffff69', borderTopLeftRadius: 20, borderTopRightRadius: 20 }}>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '700', fontSize: 22, marginTop: 40 }}>Signup</Text>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 40 }}>User Name</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            secureTextEntry={false}
                            onChange={(event: any) => setUserName(event.target.value)}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="User Name"
                        />
                    </View>
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Mobile Number</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            secureTextEntry={false}
                            onChange={(event: any) => setMobile(event.target.value)}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="077xxxxxxx"
                        />
                    </View>
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Password</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            secureTextEntry={true}
                            onChange={(event: any) => setPassword(event.target.value)}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="Password"
                        />
                    </View>
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Confirm Password</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            secureTextEntry={true}
                            onChange={(event: any) => setConfirmPassword(event.target.value)}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="Confirem Password"
                        />
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 15 }}>
                    <Pressable onPress={() => navigation.navigate('Login')}>
                        <Text style={{ color: '#4a4141', fontWeight: '500', fontSize: 15 }}>Do you have an account? <b>Sign in</b></Text>
                    </Pressable>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginTop: 10, marginBottom: 30 }}>
                    <ButtonSignUp buttonText="Sign up"></ButtonSignUp>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    linearGradient: {
        flex: 1,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        height: '100%',
    },
});

