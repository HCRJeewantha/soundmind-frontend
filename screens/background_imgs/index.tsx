export const SUNSET = require('./sunset.jpg'); 
export const LOGO = require('./logo.png'); 