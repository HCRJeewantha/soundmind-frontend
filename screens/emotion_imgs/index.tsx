export const NEUTRAL = require('./neutral.png'); 
export const HAPPY = require('./happy.png'); 
export const SAD = require('./sad.png'); 
export const FEAR = require('./fear.png'); 
export const ANGRY = require('./angry.png'); 
