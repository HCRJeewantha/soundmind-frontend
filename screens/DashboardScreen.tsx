import React, { useState, useEffect } from "react";
import { Pressable, StyleSheet, TextInput, Text, Image, ScrollView, Dimensions, FlatList } from "react-native";
import { View } from "../components/Themed";
import { RootStackScreenProps, RootTabScreenProps } from "../types";
import { LinearGradient } from 'expo-linear-gradient';
import { LOGO, PLAYLIST } from "./background_imgs";
import { FontAwesome } from "@expo/vector-icons";
import axios from 'axios'

const baseUrl = 'http://127.0.0.1:8000';
const { width } = Dimensions.get('window');

export default function TabFourScreen({ navigation }: RootTabScreenProps<'TabFour'>) {

    const [musicForYou, setMusicForYou] = useState([]);

    const getMusicForYou = async () => {
        axios.get(
            baseUrl + '/get-music-for-you/rock'
        ).then((response) => {
            console.log(response.data.tracks.track)
            setMusicForYou(response.data.tracks.track)
        }).catch((error) => {
            console.log(error)
        })
    }

    useEffect(() => {
        //get playlist
        getMusicForYou()
    }, []);

    const renderItem = ({ item, index }: any) => (
        <View>
            <View style={styles.view} />
        </View>
    );


    return (
        <React.Fragment>
            <LinearGradient
                colors={['#262626', '#75dadf']}
                style={styles.background}
            />
            <ScrollView style={styles.container}>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', width: '95%', height: '80px', backgroundColor: 'none' }}>
                        <View style={{ width: '90%', height: '50px', padding: 5, justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>Good Evening</Text>
                        </View>
                        <View style={{ width: '10%', height: '50px', padding: 5, justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }} >
                            <FontAwesome
                                style={{ marginRight: 5 }}
                                name="gear"
                                size={20}
                                color={'#fff'}
                            />
                        </View>
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', backgroundColor: '#00000052', width: '95%', borderRadius: 10 }}>
                        <View style={{ width: '30%', height: 'auto', justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <View style={{ height: 80, width: 90, alignContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                                <Image
                                    style={{
                                        borderTopLeftRadius: 10,
                                        borderBottomLeftRadius: 10,
                                        width: '100%',
                                        height: '100%',
                                        resizeMode: 'stretch'
                                    }}
                                    source={PLAYLIST} />
                            </View>
                        </View>
                        <View style={{ width: '90%', height: 'auto', padding: 5, justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>My Playlist</Text>
                        </View>
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', width: '95%', backgroundColor: 'none' }}>
                        <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>Music For You</Text>
                    </View>
                </View>
                <FlatList
                    ref={(scrollView) => { scrollView = scrollView; }}
                    style={styles.container}
                    horizontal={true}
                    decelerationRate={0}
                    snapToInterval={width - 60}
                    snapToAlignment={"center"}
                    data={musicForYou}
                    renderItem={renderItem}
                    contentInset={{
                        top: 0,
                        left: 30,
                        bottom: 0,
                        right: 30,
                    }} />
            </ScrollView>
        </React.Fragment>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    linearGradient: {
        flex: 1,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        height: '100%',
    },
    view: {
        backgroundColor: 'blue',
        width: width - 80,
        margin: 10,
        height: 200,
        borderRadius: 10,
    },
    view2: {
        backgroundColor: 'red',
        width: width - 80,
        margin: 10,
        height: 200,
        borderRadius: 10,
    },
});

