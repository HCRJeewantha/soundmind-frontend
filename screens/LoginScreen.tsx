import React, { useState } from "react";
import { Pressable, StyleSheet, TextInput, Text, Image } from "react-native";
import { View } from "../components/Themed";
import { LinearGradient } from 'expo-linear-gradient';
import { LOGO } from "./background_imgs";
import axios from 'axios'
import Toast from 'react-native-toast-message';

//Base URL
const baseUrl = 'http://127.0.0.1:8000';

export function LoginScreen({ navigation }: any) {

    const [mobile, setMobile] = useState('');
    const [password, setPassword] = useState('');

    const showToast = (title: any, message: any, type: any) => {
        Toast.show({
            type: type,
            text1: title,
            text2: message
        });
    }

    const validations = () => {
        let formIsValid = true;
        if (!mobile) {
            showToast("Error", "Mobile Number is required", "error")
            formIsValid = false;
        }
        if (mobile.length > 11) {
            showToast("Error", "Mobile Number shoud less than 10", "error")
            formIsValid = false;
        }
        if (!password) {
            showToast("Error", "Password required", "error")
            formIsValid = false;
        }
        return formIsValid;
    }

    const signIn = async () => {
        if (validations()) {
            const payload = {
                "mobile": mobile,
                "password": password
            }

            axios.post(
                baseUrl + '/sign-in',
                payload
            ).then((response) => {
                localStorage.setItem("id", response.data.id)
                localStorage.setItem("name", response.data.name)
                showToast("Success", "Login Successful", "success")
                navigation.navigate('Welcome')
            }).catch((error) => {
                if (error.response) {
                    showToast("Error", error.response.data.detail, "error")
                }
            });
        }
    };

    return (
        <View style={styles.container}>
            <LinearGradient
                colors={['#262626', '#75dadf']}
                style={styles.background}
            />
            <View style={{ width: '100%', height: 'auto', bottom: 0, position: 'absolute', backgroundColor: '#ffffff69', borderTopLeftRadius: 20, borderTopRightRadius: 20 }}>

                <View style={{ height: 120, width: 120, borderRadius: 100, alignContent: 'center', alignItems: 'center', margin: 'auto', zIndex: 1, top: -50, backgroundColor: '#ffffff69' }}>
                    <Image
                        style={{
                            width: '100%',
                            height: '100%',
                            resizeMode: 'stretch'
                        }}
                        source={LOGO} />
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '700', fontSize: 22, marginTop: 40 }}>Signin</Text>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Mobile Number</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            onChange={(event: any) => setMobile(event.target.value)}
                            secureTextEntry={false}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="07xxxxxx"
                        />
                    </View>
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Password</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            onChange={(event: any) => setPassword(event.target.value)}
                            secureTextEntry={true}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="password"
                        />
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 15 }}>
                    <Pressable onPress={() => navigation.navigate('Registration')}>
                        <Text style={{ color: '#4a4141', fontWeight: '500', fontSize: 15 }}>Don't you have an account? <b>Sign up</b></Text>
                    </Pressable>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginTop: 10, marginBottom: 40 }}>
                    <Pressable
                        testID="signInBtn"
                        style={{
                            alignItems: 'center',
                            justifyContent: 'center',
                            width: '100%',
                        }}
                        onPress={() => signIn()}>
                        <LinearGradient
                            colors={['#50aeffeb', '#26ced7f2']}
                            style={{
                                marginTop: 30,
                                borderRadius: 10,
                                borderWidth: 1,
                                borderColor: '#2196f3',
                                alignItems: 'center',
                                justifyContent: 'center',
                                width: '80%',
                                height: '50px',
                                backgroundColor: '#2196f3',
                                shadowOpacity: 0.25,
                                shadowRadius: 30,
                                shadowOffset: {
                                    height: 0,
                                    width: 0,
                                },
                            }}>
                            <Text style={{ color: '#fff', fontSize: 20, fontWeight: '600' }}>Sign In</Text>
                        </LinearGradient>
                    </Pressable>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    linearGradient: {
        flex: 1,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        height: '100%',
    },
});

