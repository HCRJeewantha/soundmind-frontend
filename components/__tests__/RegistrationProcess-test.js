import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { RegistrationScreen } from '../../screens/RegistrationScreen';
import renderer from 'react-test-renderer';
import 'isomorphic-fetch';

describe('Testing search song by name', () => {

     /*
    Snapshot Testing
    */
    test("Signin UI render correctly", async () => {
        const tree = renderer.create(<RegistrationScreen/>).toJSON();
        expect(tree).toMatchSnapshot();
    });


    /*
    Intergration Testing
    */
    test("Signup UI render correctly", async () => {
        const { getByText, getByPlaceholderText } = render(
            <RegistrationScreen />
        );
        fireEvent.changeText(getByPlaceholderText('077xxxxxxx'), '0771234567');
        fireEvent.changeText(getByPlaceholderText('Password'), 'Ravindu@123');
        fireEvent.changeText(getByPlaceholderText('Confirem Password'), 'Ravindu@123');
        fireEvent.press(getByText("Sign up"));
        // const response  =  Services.signIn('0774133985','saf');
    });

});