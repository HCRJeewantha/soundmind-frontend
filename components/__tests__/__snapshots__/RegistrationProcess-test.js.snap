// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing search song by name Signin UI render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4280690214,
          4285913823,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "backgroundColor": "#ffffff69",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "bottom": 0,
          "height": "auto",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#4a4141",
          "fontSize": 22,
          "fontWeight": "700",
          "marginLeft": "10%",
          "marginTop": 40,
        }
      }
    >
      Signup
    </Text>
    <Text
      style={
        Object {
          "color": "#4a4141",
          "fontSize": 15,
          "fontWeight": "600",
          "marginLeft": "10%",
          "marginTop": 40,
        }
      }
    >
      User Name
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 10,
              "height": "50px",
              "marginTop": "10px",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 30,
              "width": "80%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChange={[Function]}
          placeholder="User Name"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Object {
              "height": "100%",
              "padding": "10px",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#4a4141",
          "fontSize": 15,
          "fontWeight": "600",
          "marginLeft": "10%",
          "marginTop": 20,
        }
      }
    >
      Mobile Number
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 10,
              "height": "50px",
              "marginTop": "10px",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 30,
              "width": "80%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChange={[Function]}
          placeholder="077xxxxxxx"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Object {
              "height": "100%",
              "padding": "10px",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#4a4141",
          "fontSize": 15,
          "fontWeight": "600",
          "marginLeft": "10%",
          "marginTop": 20,
        }
      }
    >
      Password
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 10,
              "height": "50px",
              "marginTop": "10px",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 30,
              "width": "80%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChange={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "height": "100%",
              "padding": "10px",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#4a4141",
          "fontSize": 15,
          "fontWeight": "600",
          "marginLeft": "10%",
          "marginTop": 20,
        }
      }
    >
      Confirm Password
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderRadius": 10,
              "height": "50px",
              "marginTop": "10px",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 30,
              "width": "80%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChange={[Function]}
          placeholder="Confirem Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "height": "100%",
              "padding": "10px",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "marginVertical": 15,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#4a4141",
              "fontSize": 15,
              "fontWeight": "500",
            }
          }
        >
          Do you have an account? 
          <b>
            Sign in
          </b>
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "none",
            "height": "auto",
            "justifyContent": "center",
            "marginBottom": 30,
            "marginTop": 10,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                3947933439,
                4062629591,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196f3",
              "borderColor": "#2196f3",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "50px",
              "justifyContent": "center",
              "marginTop": 30,
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 30,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "600",
              }
            }
          >
            Sign up
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;
